src/reflect/scala/reflect/internal/Types.scala

```diff
@@ -3131,13 +3131,47 @@ trait Types
        */
       def unifyFull(tpe: Type): Boolean = {
         def unifySpecific(tp: Type) = {
-          sameLength(typeArgs, tp.typeArgs) && {
+          if(sameLength(typeArgs, tp.typeArgs)) {
```

Comment:
This is `compareLengths(typeArgs, tp.typeArgs) == 0`. Let's store `compareLengths`'s result and reuse it in the `else if` below. IntelliJ says there are 6 occurrences of tp.typeArgs, so let's give that its own local.